### DISCUSSION TITLE
What NPM packages would you like to be able to use in BWE?

### DISCUSSION THREAD

**Discussion by mpeterdev** [near/react-on-chain#166] [Announcements]
The BWE team is looking to compile a list of NPM packages which developers want to be able to use in BOS components so that we can test their compatibility. Please chime in with packages which you want to use and reply/react to posts from others mentioning packages you would use as well

**Comment by mpeterdev** [comment]
markdown rendering:
https://www.npmjs.com/package/react-markdown
https://www.npmjs.com/package/marked-react

**Comment by mpeterdev** [comment]
[radix primitives](https://www.radix-ui.com/primitives/docs/overview/introduction)

**Comment by mpeterdev** [comment]
[TanStack Query](https://tanstack.com/query/latest/docs/react/overview) (formerly known as React Query)

**Comment by mpeterdev** [comment]
[Zustand](https://github.com/pmndrs/zustand) for state management

**Comment by mpeterdev** [comment]
[lodash](https://lodash.com/)

**Comment by mpeterdev** [comment]
[react-syntax-highlighter](https://www.npmjs.com/package/react-syntax-highlighter) for code snippets

**Comment by mpeterdev** [comment]
[shadcn/ui](https://ui.shadcn.com/)

**Comment by calebjacob** [comment]
[jsdiff](https://github.com/kpdecker/jsdiff) for displaying component commit history and diffs

**Comment by charleslavon** [comment]
[idos-sdk](https://www.npmjs.com/package/@idos-network/idos-sdk) for proof-of-humanness & sovereign identity

**Comment by elliotBraem** [comment]
https://zod.dev/

**Comment by elliotBraem** [comment]
https://www.react-hook-form.com/

**Comment by ALuhning** [comment]
GraphQL support - [Apollo Client ](https://www.npmjs.com/package/@apollo/client)

**Comment by Megha-Dev-19** [comment]
Drag and Drop -[ react-beautiful-dnd](https://www.npmjs.com/package/react-beautiful-dnd)

**Comment by Canvinus** [comment]
[Chakra UI](https://chakra-ui.com/)

**Comment by igorperic17** [comment]
echarts - https://www.npmjs.com/package/echarts

**Comment by mikedotexe** [comment]
This is something being used on Farcaster Frames, and think it could be useful when you want to make small little graphs or other elements in a scalable SVG way:
https://www.npmjs.com/package/satori

More info from the Frames technical document: https://warpcast.notion.site/Farcaster-Frames-4bd47fe97dc74a42a48d3a234636d8c5

**Comment by mikedotexe** [comment]
This is a nice-to-have, but thought it would be cool to have CSS keyframes, and I think we might need:
https://www.npmjs.com/package/styled-components

**Comment by calebjacob** [comment]
Query API heavily relies on GraphQL - would be sweet if this library could work: https://gql-tada.0no.co/
